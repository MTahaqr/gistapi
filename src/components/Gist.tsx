import React from 'react';

interface GistProps {
  gist: string;
}

const Gist = ({ gist }: GistProps) => <></>;

export default Gist;
