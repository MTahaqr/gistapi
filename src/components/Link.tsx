import React from 'react';
// Packages
import styled from 'styled-components';

interface LinkProps {
  href: string;
  target: string;
  rel?: string;
  children: React.ReactNode;
}

const Link: React.FC<LinkProps> = ({ href, target, rel = 'noreferrer', children }) => {
      /**
   * @Render
   */
  return (
    <LinkWrapper href={href} target={target} rel={rel}>
      {children}
    </LinkWrapper>
  );
};

export { Link, LinkProps };

const LinkWrapper = styled.a`
  cursor: pointer;
  text-decoration: none;
`;
