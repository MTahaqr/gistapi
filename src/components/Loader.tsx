import React from 'react';
// Packages
import { RotateSpinner } from 'react-spinners-kit';
import styled from 'styled-components';

interface LoaderProp {
  color?: string;
}
const Loader: React.FC<LoaderProp> = ({ color = '#24292e' }) => {
  /**
   * @Render
   */
  return (
    <StyledContainer>
      <RotateSpinner color={color} />
    </StyledContainer>
  );
};

const StyledContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

export { Loader, LoaderProp };
