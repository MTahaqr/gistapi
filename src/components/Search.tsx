import React, { ChangeEvent, useEffect, useState } from 'react';
// Packages
import styled from 'styled-components';
// Hooks
import useDebounce from '../Hooks/useDebounce';
// Packages
import { AiOutlineSearch } from 'react-icons/ai';

interface SearchProps {
  onChange: (value: string) => void;
}
const Search = ({ onChange }: SearchProps) => {
  const [searchText, setSearchText] = useState<string>('');
  /**
   * @Methods
   */
  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    setSearchText(event.target.value || ''.trim());
  };

  const debouncedSearchVal = useDebounce(searchText, 500);

  /**
   * @Effects
   */
  useEffect(() => {
    onChange(debouncedSearchVal);
  }, [debouncedSearchVal, onChange]);

  /**
   * @Render
   */
  return (
    <Wrapper>
      <InputBox>
        <AiOutlineSearch color='black' size={20} />
        <Input
          value={searchText}
          placeholder='Search Gists for the username'
          onChange={handleChange}
        />
      </InputBox>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  padding: 8px;
  background-color: #ffffff;
  font-size: 14px;
  line-height: 1.5;
  border-radius: 6px;
  margin: 0 16px;
`;

const InputBox = styled.div`
  border-radius: 4px;
  display: flex;
  width: 400px;
`;

const Input = styled.input`
  border: none;
  width: 100%;
  font-size: 16px;

  &:focus {
    outline: 0;
  }
`;

export { Search, SearchProps };
