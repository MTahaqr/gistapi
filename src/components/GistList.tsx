import React, { useEffect, useState } from 'react';
import { GistResult } from '../services/gistService';
import styled from 'styled-components';
import { Gist } from './Gist';

const GistList = ({ gistList }: { gistList: GistResult[] }) => {
  return (
    <StyledContainer>
      {gistList?.map((data, index) => (
        <Gist key={index} {...data} />
      ))}
    </StyledContainer>
  );
};

export { GistList };

const StyledContainer = styled.div``;
