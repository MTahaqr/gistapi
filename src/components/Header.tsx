import React from 'react';
import styled from 'styled-components';
// eslint-disable-next-line
// @ts-ignore skipping next line because no type was available for react-octicon library
// import Octicon from 'react-octicon';
import { Search } from './Search';
import { BsGithub } from 'react-icons/bs';

const Wrapper = styled.div`
  background-color: #24292e;
  position: sticky;
  top: 0px;
  width: 100%;
  color: #ffffff;
  z-index: 32;
  padding: 16px;
  font-size: 14px;
  line-height: 1.5;
  display: flex;
  align-items: center;
`;

interface HeaderProps {
  onChange: (value: string) => void;
}
const Header: React.FC<HeaderProps> = ({ onChange }) => {
  return (
    <Wrapper>
      <BsGithub size={32} />
      <Search onChange={onChange} />
    </Wrapper>
  );
};

export { Header, HeaderProps };
