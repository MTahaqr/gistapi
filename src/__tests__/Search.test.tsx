import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { Search, SearchProps } from '../components/Search';

describe('Search', (): void => {
  let wrapper: ShallowWrapper;
  beforeEach((): void => {
    const props: SearchProps = {
      onChange: jest.fn(),
    };
    wrapper = shallow(<Search {...props} />);
  });

  it('should render properly', (): void => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
