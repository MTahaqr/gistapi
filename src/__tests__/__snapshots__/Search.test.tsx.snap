// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render properly 1`] = `
<styled.div>
  <styled.div>
    <AiOutlineSearch
      color="black"
      size={20}
    />
    <styled.input
      onChange={[Function]}
      placeholder="Search Gists for the username"
      value=""
    />
  </styled.div>
</styled.div>
`;
